providers/dns/bluecat/internal/client.go

```diff
@@ -0,0 +1,286 @@
+		return fmt.Errorf("status code: %d, message: %s", resp.StatusCode, string(data))
+	}
+
+	authBytes, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return err
+	}
+
+	authResp := string(authBytes)
+	if !strings.Contains(authResp, "successfully") {
+		msg := strings.Trim(authResp, "\"")
+		return fmt.Errorf("request failed to delete session: %s", msg)
+	}
+
+	c.token = ""
+
+	return nil
+}
+
+// Deploy the DNS config for the specified entity to the authoritative servers.
+// https://docs.bluecatnetworks.com/r/Address-Manager-API-Guide/POST/v1/quickDeploy/9.1.0
+func (c *Client) Deploy(entityID uint) error {
+	queryArgs := map[string]string{
+		"entityId": strconv.FormatUint(uint64(entityID), 10),
+	}
+
+	resp, err := c.sendRequest(http.MethodPost, "quickDeploy", nil, queryArgs)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusCreated {
+		data, _ := io.ReadAll(resp.Body)
+		return fmt.Errorf("status code: %d, message: %s", resp.StatusCode, string(data))
+	}
+
+	return nil
+}
+
+// AddEntity A generic method for adding configurations, DNS zones, and DNS resource records.
+// https://docs.bluecatnetworks.com/r/Address-Manager-API-Guide/POST/v1/addEntity/9.1.0
+func (c *Client) AddEntity(parentID uint, entity Entity) (uint64, error) {
+	queryArgs := map[string]string{
+		"parentId": strconv.FormatUint(uint64(parentID), 10),
+	}
+
+	resp, err := c.sendRequest(http.MethodPost, "addEntity", entity, queryArgs)
+	if err != nil {
+		return 0, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		data, _ := io.ReadAll(resp.Body)
+		return 0, fmt.Errorf("status code: %d, message: %s", resp.StatusCode, string(data))
+	}
+
+	addTxtBytes, _ := io.ReadAll(resp.Body)
+
+	// addEntity responds only with body text containing the ID of the created record
+	addTxtResp := string(addTxtBytes)
+	id, err := strconv.ParseUint(addTxtResp, 10, 64)
+	if err != nil {
+		return 0, fmt.Errorf("addEntity request failed: %s", addTxtResp)
+	}
+
+	return id, nil
+}
+
+// GetEntityByName Returns objects from the database referenced by their database ID and with its properties fields populated.
+// https://docs.bluecatnetworks.com/r/Address-Manager-API-Guide/GET/v1/getEntityById/9.1.0
+func (c *Client) GetEntityByName(parentID uint, name, objType string) (*EntityResponse, error) {
+	queryArgs := map[string]string{
+		"parentId": strconv.FormatUint(uint64(parentID), 10),
+		"name":     name,
+		"type":     objType,
+	}
+
+	resp, err := c.sendRequest(http.MethodGet, "getEntityByName", nil, queryArgs)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		data, _ := io.ReadAll(resp.Body)
+		return nil, fmt.Errorf("status code: %d, message: %s", resp.StatusCode, string(data))
+	}
+
+	var txtRec EntityResponse
+	err = json.NewDecoder(resp.Body).Decode(&txtRec)
+	if err != nil {
+		return nil, fmt.Errorf("JSON decode: %w", err)
+	}
+
+	return &txtRec, err
```

Comment:
For the sake of consistency, this should be:

```suggestion
	if err = json.NewDecoder(resp.Body).Decode(&txtRec); err != nil {
		return nil, fmt.Errorf("JSON decode: %w", err)
	}

	return &txtRec, nil
```